internal/config/state_test.go

```diff
@@ -71,14 +71,31 @@ func TestLibrarianState_Validate(t *testing.T) {
 			wantErr:    true,
 			wantErrMsg: "libraries cannot be empty",
 		},
+		{
+			name: "duplicate library IDs",
+			state: &LibrarianState{
+				Image: "gcr.io/test/image:v1.2.3",
+				Libraries: []*LibraryState{
+					{
+						ID:          "x",
+						SourceRoots: []string{"src/x"},
+					},
+					{
+						ID:          "x",
+						SourceRoots: []string{"src/x"},
+					},
+				},
+			},
+			wantErr:    true,
+			wantErrMsg: "duplicate library ID",
+		},
 	} {
 		t.Run(test.name, func(t *testing.T) {
 			err := test.state.Validate()
 			if test.wantErr {
 				if err == nil {
 					t.Error("Librarian.Validate() should fail")
-				}
-				if !strings.Contains(err.Error(), test.wantErrMsg) {
+				} else if !strings.Contains(err.Error(), test.wantErrMsg) {
```

Comment:
Is this `else` necessary?

Reply:
I think a `if-else` is not as straightforward as multiple `if`.

@codyoss what do you think?

Reply:
I don't think this is a blocking issue for this PR.

We can decide the test format later and raise a dedicated PR.